### PR TITLE
Support using discord channel ids

### PR DESCRIPTION
**Pull request by mraof**

### CODE REVIEW

**Comment by ekmartin** [comment]
Would you care to explain why you think is useful as well? Can't see any immediate gain, but I might be missing something.

**Comment by mraof** [comment]
It allows the config to keep working even if discord channels are renamed, which is the problem I was having.
It might also allow specifying which channel if the bot is in multiple servers that have a channel named the same, but I haven't tested that

File: lib/bot.js [hunk 1]
```diff
@@ -193,9 +194,9 @@ class Bot {
     const discordChannelName = this.invertedMapping[channel.toLowerCase()];
     if (discordChannelName) {
       // #channel -> channel before retrieving and select only text channels:
-      const discordChannel = this.discord.channels
+      const discordChannel = discordChannelName.charAt(0) === '#' ? this.discord.channels
```
Comment (ekmartin) [inline lib/bot.js]: `discordChannelName.startsWith('#')`

File: lib/bot.js [hunk 2]
```diff
@@ -154,7 +154,8 @@ class Bot {
     if (author.id === this.discord.user.id) return;
 
     const channelName = `#${message.channel.name}`;
-    const ircChannel = this.channelMapping[channelName];
+    const ircChannel = this.channelMapping[message.channel.id in this.channelMapping ?
```
Comment (ekmartin) [inline lib/bot.js]: `const ircChannel = this.channelMapping[message.channel.id] || this.channelMapping[channelName]` maybe?

**Comment by ekmartin** [comment]
Ye, that makes sense. Care to add a test or two?

**Comment by mraof** [comment]
I'm afraid I don't actually know how to add tests

**Comment by Sanqui** [comment]
I just ran an issue when joining the bot onto two Discord servers with #﻿general.  Would appreciate this a lot.  Might give writing tests a shot but I'm not familiar with it at all.

**Comment by Throne3d** [comment]
I've had a go at adding tests for this in PR #202. Thanks for putting in the work for the functionality.

**Comment by ekmartin** [comment]
Closing this since #202 was merged. Thanks!

**Pull request merged**